.cbmc-batch/stubs/aws_byte_cursor_read_be16_override.c

```diff
@@ -0,0 +1,56 @@
+/*
+ * Copyright 2019 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"). You may not use
+ * this file except in compliance with the License. A copy of the License is
+ * located at
+ *
+ *     http://aws.amazon.com/apache2.0/
+ *
+ * or in the "license" file accompanying this file. This file is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or
+ * implied. See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include <aws/common/byte_buf.h>
+#include <proof_helpers/nondet.h>
+
+/**
+ * The signature for the value generator, if it is used.
+ */
+#ifdef AWS_BYTE_CURSOR_READ_BE16_GENERATOR
+uint16_t AWS_BYTE_CURSOR_READ_BE16_GENERATOR(const struct aws_byte_cursor *cursor);
+#endif
+
+/**
+ * This function is used in deserilizing values from a byte cursor.
+ * Sometimes, for the proof to go through, it is expected that certain values in byte stream are costrained.
+ * For example, in the aws_cryptosdk_enc_ctx_deserilize() proof, the first value we read is the number of elements,
+ * which we need to be constrained in order to ensure that the proof finishes. All other values can be left nondet.
+ * If there is no structure that must be followed, the function can be left undefined, and CBMC will automatically
+ * give nondet answers.
```

Comment:
CBMC will actually leave the arguments untouched (unless you explicitly generated function bodies that modify pointer-typed arguments) - so, yes, CBMC will return a non-deterministic value, but the "value" that you really care about is to be stored in one of the arguments.

Reply:
done